src/platform_impl/macos/window.rs

```diff
@@ -977,6 +981,33 @@ impl UnownedWindow {
         }
     }
 
+    #[inline]
+    pub fn request_user_attention(&self, request_type: Option<RequestUserAttentionType>) {
+        let mut shared_state = self.shared_state.lock().unwrap();
+
+        unsafe {
+            match request_type {
+                Some(request_type) => {
+                    shared_state.request_user_attention_id =
+                        Some(NSApp().requestUserAttention_(match request_type {
+                            RequestUserAttentionType::Critical => {
+                                NSRequestUserAttentionType::NSCriticalRequest
+                            }
+                            RequestUserAttentionType::Informational => {
+                                NSRequestUserAttentionType::NSInformationalRequest
+                            }
```

Comment:
Could you `map` request_type to the corresponding platform variants and not have inline match?